examples/retrieval/retrieval.cpp

```diff
@@ -0,0 +1,278 @@
+#include "common.h"
+#include "llama.h"
+
+#include <algorithm>
+#include <fstream>
+
+struct chunk {
+    // filename
+    std::string filename;
+    // original file position
+    int64_t filepos;
+    // original text data
+    std::string textdata = "";
+    // tokenized text data
+    std::vector<std::int32_t> tokens;
```

Comment:
```suggestion
    std::vector<llama_token> tokens;
```

Also remember to clear this vector, because you don't need to keep `tokens` once embeddings is calculated.